internal/testsdetection/testsdetection_test.go

```diff
@@ -0,0 +1,87 @@
+package testsdetection_test
+
+import (
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/testsdetection"
+)
+
+var coverDir string
+
+func TestMustBeTestingInTests(t *testing.T) {
+	t.Parallel()
+
+	defer func() {
+		r := recover()
+		require.Nil(t, r, "MustBeTesting should not panic as we are running in tests")
+	}()
+
+	testsdetection.MustBeTesting()
+}
+
+func TestMustBeTestingInProcess(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		integrationtestsTag bool
+
+		wantPanic bool
+	}{
+		"Pass when called in an integration tests build": {integrationtestsTag: true, wantPanic: false},
+
+		"Error (panics) when called in non tests and no integration tests": {integrationtestsTag: false, wantPanic: true},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			temp := t.TempDir()
+			testBinary := filepath.Join(temp, "testbin")
+
+			buildCmd := []string{"build", "-o", testBinary}
+			if tc.integrationtestsTag {
+				buildCmd = append(buildCmd, "-tags=integrationtests")
+			}
+			env := os.Environ()
+			if coverDir != "" {
+				buildCmd = append(buildCmd, "-cover")
+				env = append(env, "GOCOVERDIR="+coverDir)
```

Comment:
Nit, but we've a testutils definition to handle this no? (`AppendCovEnv` or `CoverDirEnv`)

Or do we want to keep tests here independent?

Reply:
oh, good ones, I didn’t know we already had helpers for those, I will use them in a followup PR